build: add a testclean target

danbev opened the pull request:
This commit adds a target named `testclean` to allow for cleaning the
temporary files generated during a test run without having to use the `clean` target.

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

danbev commented:
> Can you add this to `vcbuild.bat` too?

I'll take a look at adding this to `vcbuild.bat` today. Thanks

Trott commented:
> > Can you add this to `vcbuild.bat` too?
> 
> I'll take a look at adding this to `vcbuild.bat` today. Thanks

Looks like `vcbuild.bat` never cleans tmp directories, even if `vcbuild.bat clean` is called, so this would be adding new additional functionality to `vcbuild.bat` beyond what is being added to `Makefile` here.  I'm not even sure that we put the `tmp` directories in the same place on Windows as we do on POSIX. I'm OK leaving this off. Someone can implement it later if it's desirable to have on Windows too.

Trott commented:
> I'm not even sure that we put the `tmp` directories in the same place on Windows as we do on POSIX. 

(We do.)

danbev commented:
@Trott I've added a testclean target which can be called using:
```console
danbev@win2 MINGW64 ~/node (master)
$ ./vcbuild.bat testclean
deleting test/.tmp*
```
Sorry this took so long, I had to set up a windows vm on azure as I had deleted my local window vm due to resource constraints.

Trott commented:
Landed in a890771cd0a31bda055fc71741ace7822bc678dd
